File: 6. Python/EX42.py
def protecao(texto):
    while True:
        try:
            return int(input(texto))  # Lê e retorna um número inteiro
        except Exception:
            print("# Entrada Invalida !")

def quantidade():
    while True:
        try:
            n = int(input("# Quantos numeros voce vai digitar: "))  # Lê a quantidade de números
            if 0 < n <= 10:  # Verifica intervalo
                return n
        except Exception:
            print("# Entrada Invalida !")

def programa():
    n = quantidade()  # Obtém a quantidade
    vet = [0] * n

    for i in range(n):
        vet[i] = protecao("# Digite um numero: ")  # Preenche a lista

    print("\n# Numeros negativos")  # Exibe negativos
    for i in range(n):
        if vet[i] < 0:
            print(f"* {vet[i]}")  # Imprime número negativo

programa()  # Executa o programa

File: 4. Java/EX42.java
package Apostila;

import java.util.Scanner;

public class EX42
{
    public static void main(String[] args) 
    {
        // Variável para quantidade de números a serem digitados
        int n;
        
        // Inicializa o Scanner para entrada de dados
        Scanner ent = new Scanner(System.in);
                
        // Solicita a quantidade de números
        System.out.print("# Quantos numeros voce vai digitar: ");
        n = ent.nextInt();
        
        // Declara o vetor com tamanho n
        int vet[] = new int[n];
        
        // Garante que n não seja maior que 10
        while (n > 10) 
        {
            System.out.println("# Maximo ate 10 numeros");
            
            // Solicita novamente a quantidade de números
            System.out.print("# Quantos numeros voce vai digitar: ");
            n = ent.nextInt();
        }
        
        // Recebe os números digitados
        for (int i = 0; i < n; i++)
        {
            System.out.print("# Digite um numero: ");
            vet[i] = ent.nextInt();
        }
        
        // Exibe os números negativos
        System.out.println("\n# Numeros negativos");
        
        for (int i = 0; i < n; i++)
        {
            // Verifica e exibe os números negativos
            if (vet[i] < 0)
            {
                System.out.println("* " + vet[i]);                
            }
        }        
    }
}
